sqlmesh/core/config/scheduler.py

```diff
@@ -69,6 +84,22 @@ def create_state_sync(self, context: Context) -> StateSync:
         schema = context.config.get_state_schema(context.gateway)
         return EngineAdapterStateSync(engine_adapter, schema=schema, console=context.console)
 
+    def get_default_catalog(self, context: Context) -> t.Optional[str]:
```

Comment:
what's the point of this? why not just set the engine_adapter current catalog in the setter?

Reply:
setter doesn't have access to engine adapter

Reply:
what do you mean? just set it whenever the catalog gets overridden unlike the lazy style  implementation you did here